### PR TITLE
Update NSIS 3.09 to 3.10

### PR DESCRIPTION
**Pull request by dpy013**
### Summary of the issue:
[pull](https://github.com/nvaccess/NSIS-build/pull/4)
### Description of user facing changes
Updating NSIS
### Description of development approach
none
### Testing strategy:
Requires a test compilation build
### Known issues with pull request:
none
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [ ] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
  - Updated Unicode CLDR for improved language support.
  - Enhanced NVDA screen reader reporting for better accessibility.
  - Improved Python console behavior for a smoother user experience.
  - Upgraded NSIS installer from version 3.09 to 3.10 for better performance and stability.

### CODE REVIEW

**Comment by dpy013** [comment]
@coderabbitai review

**Comment by seanbudd** [comment]
@dpy013 - have you tested this in NVDA yet with a build?

I've checked the files changed and everything looks ok.

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Unicode CLDR has been updated.
 * NVDA will now report figures with no accessible children, but with a label or description. (#14514)
 * When reading by line in browse mode, "caption" is no longer reported on each line of a long figure or table caption. (#14874)
 * In the Python console, the last unexecuted command will no longer be lost when moving in the input history. (#16653, @CyrilleB79)
+  * Updated NSIS 3.09 to 3.10 (#16674, @dpy013)
```
Comment (seanbudd) [inline user_docs/en/changes.md:34]: this should be in "component updates"

**Comment by dpy013** [comment]
> @dpy013 - have you tested this in NVDA yet with a build?
> 
> I've checked the files changed and everything looks ok.

Currently only the portable version has been tested, because it is using the unit's computer, so only the portable version can be created for testing.
It is recommended that if you have time, you can test the part of the installation to the computer.

**Comment by seanbudd** [comment]
@dpy013 - you should be able to create an installable build with `scons launcher`. You can also use the installer generated from AppVeyor in this PR.

**Comment by dpy013** [comment]
Just used nvda_snapshot_pr16674-32353 to test the two steps of creating a portable version, and installing it to the computer, here are the two logs:
[nvda-I.log](https://github.com/user-attachments/files/15782378/nvda-I.log)
[nvda-P.log](https://github.com/user-attachments/files/15782380/nvda-P.log)

**Comment by seanbudd** [comment]
Thanks, please update the reference to the NSIS submodule and this can be merged

**Review by seanbudd — APPROVED** [review]

**Pull request merged**